### DISCUSSION TITLE
Dark theme for Google Sheets

### DISCUSSION THREAD

**Discussion by gitpro2005** [winghongchan/dark-docs#22] [Ideas]
Could you make a version of this theme for Google Sheets? I use it very often and I would like if it also had a dark theme. Thank you, your work is awesome!

**Comment by winghongchan** [comment]
We’ll see. If I start using Sheets more, then I’ll make a dark mode for it. I only make userstyles for products I use frequently so that I might be the first to notice bugs or new features requiring changes to my code (like the new dropdown menu for zooming in Google Slides). 

My [dark mode for Google Slides](https://github.com/winghongchan/dark-docs/blob/main/shades-and-slides.user.css) is maybe about a week or two from completion. After that I will consider making a sheet to track my grades in university or dig up my old workout plan to get me to use Sheets more often so I can make a good dark mode for it.

**Comment by winghongchan** [comment]
- Progress update: my dark mode for Google Slides has been released: [Shades & Slides](https://userstyles.world/style/11559/shades-slides)
- I’ve figured out how to do icon colouring if I want to make the Docs editors themeable. I now have a Google Sheet that I use for work, which means I now have motivation to get it done. 
- I can now start working on M3 Themes for Google Sheets.

**Comment by winghongchan** [comment]
Progress update: instead of M3 Themes for Google Sheets, I’ve started working on [Dynamic Docs](https://github.com/winghongchan/dark-docs/blob/main/dynamic-docs.user.css) which aims to bring Material You theming to Docs, Sheets, and Slides.

**Comment by gitpro2005** [comment]
That’s great news, really nice to hear this after a while!

I’ve read your Dynamic Docs style and it looks very professional and really promising. I love how it brings a complete, high‑fidelity Material 3 toolset, with elevation, transitions, ripple effects, and more.

Do you plan to treat it as a successor or replacement for the previous styles, or will you keep it as a separate, similar alternative that focuses on Dynamic Color themes?

I’ve also been experimenting with Material 3 and Dynamic Color in userstyles. I have my 2023 style, [Stylus M3](https://userstyles.world/style/8830/stylus-m3-material-design-3-for-stylus), which is a bit immature and currently broken, but I’m working on a V2 together with Gemini Pro and the results have been surprisingly good. In this new version, you can choose any base color for the theme, and it uses `color-mix()` (in the `oklab` color space) to approximate a Material 3‑like palette for all the standard roles. I’m planning to publish an early version as soon as I can. 

Using the same color system, I also made a Contacts style with full light and dark themes and dynamic colors based on a single source color. Besides `color-mix()`, I’ve also been looking at other approaches, like a [hue-based demo](https://userstyles.world/style/12753/hue-based-knockoff-material-you-demo), so it was really interesting to see that you used HCT and the Material Theme Generator for your themes.

Let me know how I can contribute to Dynamic Docs, I’d be happy to help. Maybe iterating on it, covering more stuff, etc.

**Reply by winghongchan** [reply]
Once it’s ready, Dynamic Docs will replace Dark Docs. This is why:
- I’m an engineering student at university. I don’t have the time to maintain multiple userstyles as big as Dark Docs. 
- Dark Docs began before Google Docs transitioned to M3. Dark Docs is full of useless legacy code and I want a fresh start. The Googlers have begun to use CSS variables, which could make Dynamic Docs much leaner since most things will be automatically recoloured. 
- Once I add a Googley theme to Dynamic Docs that matches the default Google light and dark themes, Dynamic Docs in Googley dark mode will look practically identical to Dark Docs. 

I’m interested in having more contributors for Dynamic Docs! Keep in mind that I have a policy of not using AI-generated code in my userstyles. 

Right now, my main problem with Dynamic Docs is: when bold, italic, or underline are active, the icons are broken. (They weren’t broken before.) I’d appreciate help with that. Also if you have a more efficient way to recolouring the icons (in a way that’s still accurate and scalable) please propose it. 
For now, I’m a bit busy with maintaining Dark Docs. 
Keep in mind that any code you contribute to Dynamic Docs will be licensed under GNU GPLv3.
